### PR TITLE
Add support to get data from external source

### PR DESCRIPTION
**Pull request by jonathanolsson**
Extend the routes so that the front-end can get data from a source that i specified in the plugin.

### CODE REVIEW

File: routes/routes.js [hunk 1]
```diff
@@ -11,11 +11,72 @@ var appRouter = function(app, upload) {
...
+		console.log("CALL: ");
+
+		//A JSON object that contains all data to access a remote API.
+		var options = {
+			hostname: req.body.sourceURL,
+			port: req.body.port,
+			path: req.body.path,
+			method: req.body.method
+		};
+		console.log(options);
```
Comment (jonathanolsson) [inline routes/routes.js]: What is this here for?

File: routes/routes.js [hunk 2]
```diff
@@ -11,11 +11,72 @@ var appRouter = function(app, upload) {
...
+				console.log('No more data in response.');
+			});
+		});
+
+		//Message on error.
+		require.on('error', (e) => {
+			console.log('problem with request:', e.message);
+			res.end("error: " + e.message)
+		});
+		require.end();
```
Comment (jonathanolsson) [inline routes/routes.js]: Remove this comment. This sould not end the connection!

File: app.js [hunk 3]
```diff
@@ -1,14 +1,15 @@
...
 var app = express();
 
-/*
+
 app.use(bodyParser.json());
 app.use(bodyParser.urlencoded({ extended: true }));
-var externcalls = require("./externcalls/extern.js");*/
-var routes = require("./routes/routes.js")(app, upload);
+/*var externcalls = require("./externcalls/extern.js");*/
+var routes = require("./routes/routes.js")(app, upload, http);
```
Comment (jonathanolsson) [inline app.js]: Requeire field does not require to have .js extension.

File: routes/routes.js (hunk 2 above)
Comment (jonathanolsson) [inline routes/routes.js]: Fix this

File: routes/routes.js [hunk 4]
```diff
@@ -11,11 +11,72 @@ var appRouter = function(app, upload) {
...
+		//A JSON object that contains all data to access a remote API.
+		var options = {
+			hostname: req.body.sourceURL,
+			port: req.body.port,
+			path: req.body.path,
+			method: req.body.method
+		};
+		console.log(options);
+
+		//Require data from a specified source API, the data is stored in 'chunk'.
```
Comment (paby1400) [inline routes/routes.js]: What does require mean in this sense?

**Pull request closed without merging**